src/python/vdo-setup-dmtest/source/Utils.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+"""
+  Utils - General vdo_setup_dmtest constants and utilities
+
+  Copyright (c) 2023 Red Hat
+"""
+
+from __future__ import print_function
+
+import logging
+import os
+import subprocess
+import textwrap
+import validators
+
+logger = logging.getLogger('vdo-setup-dmtest')
+INDENT = '  '
+
+def createX509Genkey(certsDir):
+  """
+  Create a x509.genkey file in the specified kernel certificate directory
+  
+  Creation is accomplished by creating the file in the current directory, and then copying it
+  to the specified certificate directory. This is done to account for python file write
+  permissons in root directories. In the case that the genkey file already exists in the
+  current directory, it will be overwritten.
+  """
+  filename = 'x509.genkey'
+  genkeyFile = os.path.join(os.getcwd(), filename)
+
+  logger.info("Creating the x509.genkey config needed for certificate/key creation")
+  try:
+    with open(genkeyFile, 'w+') as fh:
+      fh.write(textwrap.dedent("""\
+        [ req ]
+        default_bits = 4096
+        distinguished_name = req_distinguished_name
+        prompt = no
+        string_mask = utf8only
+        x509_extensions = myexts
+        [ req_distinguished_name ]
+        CN = Build time autogenerated kernel key for test modules
+        [ myexts ]
+        basicConstraints=critical,CA:FALSE
+        keyUsage=digitalSignature
+        subjectKeyIdentifier=hash
+        authorityKeyIdentifier=keyid
+        """))
+
+    result = runCommand(['sudo cp', genkeyFile, certsDir], logger)
```

Comment:
This command format doesn't match the rest of the script where you're passing things to runCommand all as a single element.

Additionally, 'sudo cp'.. shouldn't that be `'sudo', 'cp'` in this context?

Personally I'd just suggest making this match the rest of the script, but if you decide to convert everything to use this method I won't complain either.

Reply:
I don't understand. There are instances in vdo-setup-dmtest.py where runCommand is presented with a list and not a single element. I wrote runCommand so that it could handle either string or list input in order to make it adaptable.

For this context, it could be written as you suggest, or as it is. Within runCommand(), it will evaluate if the input is a string or a list and concatenate as appropriate.